store/sqlstore/channel_member_history_store.go

```diff
@@ -65,7 +65,40 @@ func (s SqlChannelMemberHistoryStore) LogLeaveEvent(userId string, channelId str
 
 func (s SqlChannelMemberHistoryStore) GetUsersInChannelDuring(startTime int64, endTime int64, channelId string) store.StoreChannel {
 	return store.Do(func(result *store.StoreResult) {
-		query := `
+		if useChannelMemberHistory, err := s.hasDataFromBefore(startTime); err != nil {
```

Comment:
The export period  is some defined range of time that we want to export data for. If data from before the start of this period exists in the ChannelMemberHistory table, then we can be sure that the export period occurred after the user upgraded to v4.5+ (when the ChannelMemberHistory table was introduced). If that's the case, then we can rely on the accurate data in this table to determine who was in the channel during the export period.

If the data doesn't contain data from before the start of the export period, then we know that at least some portion of the export period occurred before the v4.5+ upgrade, so we have to fall back to using the less accurate ChannelMembers table to determine who was in the channel

Reply:
What if your exporting from the beginning of time on a 4.5+ server? 
At least how you describe it, it won't use the table.

Reply:
The export job performs the exports in batches of 10k posts, so any batches whose start time falls before the date line will use the rough approximation method. The first batch that falls entirely after the v4.5 upgrade will use the new table. Do you have a better option in mind?

Reply:
Not commenting on the method. I am commenting on correctness. 
If I set start time to some value x, and the first join in the system is at exactly x. Then when I run this code it won't use the better method. I think it should be less than or equal too.

Reply:
Not commenting on the method. I am commenting on correctness.
If I set start time to some value x, and the first join in the system is at exactly x. Then when I run this code it won't use the better method. I think it should be less than or equal too.

Reply:
The user can't explicitly set the start time via the UI. It is taken from the system clock when they activate the Compliance Export feature. Rows will start appearing in the table just as soon as the system is updated to v4.5+ and users begin joining/leaving channels. The chances of these two timestamps lining up to be exactly equal is incredibly small. I can change the code to be <=, but it makes little difference to the operation of the feature.

Reply:
Fixed in https://github.com/mattermost/mattermost-server/pull/7951/commits/20a93313b5f860f9660d255372f891d379c7b0ff

Reply:
One final nit. The name of the function should be updated to reflect it's behavior.